src/natcap/invest/utils.py

```diff
@@ -158,10 +159,14 @@ def prepare_workspace(
             start_time = time.time()
             try:
                 yield
+            except Exception:
+                LOGGER.exception(f'Exception while executing {modelname}')
+                raise
```

Comment:
> Okay, using `LOGGER.exception` appends the Traceback to the message, right? So that explains how the Traceback gets to stdout and to the logfile. And then the `raise` is how Traceback gets to stderr.

Yep!

> I still don't understand how the Tracebacks got to stdout & the logfile before these changes, but I guess I don't need to know that now.

Well, the main way is because `cli.py` attaches a `StreamHandler(sys.stdout)` to the root logger, so as long as the logging threshold was set correctly, the traceback would appear there as well.  Part of the problem was that the logging threshold would by default prevent the traceback from showing up under certain circumstances, and I do think there may be some nondeterminism around whether the stdout log is flushed before an exception is written to stderr.  But maybe there's something else at play as well that we don't fully understand!

Reply:
> Well, the main way is because cli.py attaches a StreamHandler(sys.stdout) to the root logger, so as long as the logging threshold was set correctly, the traceback would appear there as well.

I'm afraid I still don't get it...there is `StreamHandler(sys.stdout)` on the root logger, yes, but before your changes, the CLI, nor `utils.prepare_workspace` never caught the exception to log it. And I was for sure testing the CLI with `--debug` or `-vvv` logging levels. Anyway, we can leave it a mystery because it's getting to the point where I would need to re-create it to convince myself I'm not making all this up.